components/hab/src/command/studio/enter.rs

```diff
@@ -13,19 +13,30 @@ use crate::{config,
 
 pub const ARTIFACT_PATH_ENVVAR: &str = "ARTIFACT_PATH";
 
+const AUTH_TOKEN_ENVVAR: &str = "HAB_AUTH_TOKEN";
+const BLDR_URL_ENVVAR: &str = "HAB_BLDR_URL";
+const CTL_SECRET_ENVVAR: &str = "HAB_CTL_SECRET";
 const ORIGIN_ENVVAR: &str = "HAB_ORIGIN";
 const STUDIO_CMD: &str = "hab-studio";
 const STUDIO_CMD_ENVVAR: &str = "HAB_STUDIO_BINARY";
 const STUDIO_PACKAGE_IDENT: &str = "core/hab-studio";
 
-pub fn start(ui: &mut UI, args: &[OsString]) -> Result<()> {
-    if henv::var(ORIGIN_ENVVAR).is_err() {
-        let config = config::load()?;
-        if let Some(default_origin) = config.origin {
-            debug!("Setting default origin {} via CLI config", &default_origin);
-            env::set_var("HAB_ORIGIN", default_origin);
+fn set_env_var_from_config(env_var: &str, config_val: Option<String>) {
+    if henv::var(env_var).is_err() {
+        if let Some(val) = config_val {
+            debug!("Setting {}={} via config file", env_var, val);
```

Comment:
It's helpful for other settings, but it may not be the best idea to print out _sensitive_ values like `HAB_AUTH_TOKEN` here. Perhaps adding an additional parameter to this function to control whether the value itself is printed, or just a notice that the env var was set via config file without printing the value would be a good approach.

Reply:
That's funny because I just read [this yesterday](http://www.informit.com/articles/article.aspx?p=1392524).

I think I like the [previous version](https://github.com/qubitrenegade/habitat-1/commit/fa88dfd75530b264711103e21ae6b6cc689ef0de)

```
fn set_env_var_from_config(env_var: &str, config_val: Option<String>, sensitive: Option<bool>) {
...
            match sensitive {
              Some(_) => debug!("Setting Sensitive value {} via config file", env_var),
              None => debug!("Setting {}={} via config file", env_var, val),
            }
...
```

But I think the if/else version is easier to reason about, so that's what I pushed.

Reply:
One thing I really like in rust is using simple custom `enums` instead of booleans. For example
```rust
enum Sensitivty {
    OkToPrintValue,
    DoNotPrintValue,
}
```
Allows the call site to be clearer:
```rust
    set_env_var_from_config(AUTH_TOKEN_ENVVAR, config.auth_token, Sensitivty::DoNotPrintValue);
    set_env_var_from_config(BLDR_URL_ENVVAR, config.bldr_url, Sensitivty: OkToPrintValue);
```
And then `match` ensures exhaustive handling:
```rust
            match sensitive {
                Sensitivty::DoNotPrintValue =>  debug!("Setting {}=REDACTED (sensitive) via config file", env_var)
                Sensitivty::OkToPrintValue => debug!("Setting {}={} via config file", env_var, val),
            }
```

Reply:
Ta da!  There's always a way to do what I want (well, what I _really_ want is default values/optional parameters) just couldn't quite get there.

I went with

```
enum Sensitivity {
    PrintValue,
    NoPrintValue,
}
```

Because it's a bit less typing ;)